miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/security/AuthorizationManager.java

```diff
@@ -71,15 +71,31 @@
    */
   public boolean readCheck(SecurableByProfile resource) throws IOException;
   
+  /**
+   * Checks whether the provided user has read permission on a resource
+   * 
+   * @param resource the object on which to check permission
+   * @param user the User for whom to check permission
+   * @return true if the user has permission to view this object, or if object is null; false otherwise
+   */
+  public boolean readCheck(SecurableByProfile resource, User user);
+
   /**
    * Verifies that the current user has permission to view a resource and throws an AuthorizationException if not
    * 
    * @param resource the object to check permissions for
    * @throws AuthorizationException if the current user does not have read permission to the resource
-   * @throws IOException if there is an error while looking up the current user. This does not indicate whether 
-   * the user exists or is authenticated, but represents other errors occurring while checking these conditions
    */
   public void throwIfNotReadable(SecurableByProfile resource) throws IOException, AuthorizationException;
+
+  /**
+   * Verifies that the provided user has permission to view a resource and throws an AuthorizationException if not
+   * 
+   * @param resource the object on which to check permissions
+   * @param user the User for whom to check permission
+   * @throws AuthorizationException if the current user does not have read permission to the resource
+   */
+  public void throwIfNotReadable(SecurableByProfile resource, User user) throws AuthorizationException;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'uk.ac.bbsrc.tgac.miso.service.security.AuthorizationException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)